content/news/2022-11-12-bevy-0.9/index.md

```diff
@@ -0,0 +1,1509 @@
+
+```rust
+fn some_exclusive_system(world: &mut World) { }
+
+// Old
+app.add_system(some_exclusive_system.exclusive_system())
+
+// New
+app.add_system(some_exclusive_system)
+```
+
+We've also expanded exclusive systems to support more system parameters, which vastly improves the user experience of writing exclusive systems and makes them more efficient by caching state across executions.
+
+[`SystemState`] enables using "normal" system parameters from inside an exclusive system:
+
+```rust
+// Old
+fn some_system(world: &mut World) {
+  let mut state: SystemState<(Res<Time>, Query<&mut Transform>)> =
+      SystemState::new(&mut world);
+  let (time, mut transforms) = state.get_mut(world);
+}
+
+// New
+fn some_system(world: &mut World, state: &mut SystemState<(Res<Time>, Query<&mut Transform>)>) {
+  let (time, mut transforms) = state.get_mut(world);
+}
+```
+
+[`QueryState`] enables cached access to individual queries:
+
+```rust
+// Old
+fn some_system(world: &mut World) {
+  let mut transforms = world.query::<&Transform>();
+  for transform in transforms.iter(world) {
+  }
+}
+
+// New
+fn some_system(world: &mut World, transforms: &mut QueryState<&Transform>) {
+  for transform in transforms.iter(world) {
+  }
+}
+```
+
+[`Local`] enables storing local data inside of the exclusive system:
+
+```rust
+// Old
+#[derive(Resource)]
+struct Counter(usize);
+fn some_system(world: &mut World) {
+  let mut counter = world.resource_mut::<Counter>();
+  counter.0 += 1;
+}
+
+// New
+fn some_system(world: &mut World, mut counter: Local<usize>) {
+  *counter += 1;
+}
+```
+
+[`World`]: https://docs.rs/bevy/0.9.0/bevy/ecs/world/struct.World.html
+[`System`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/trait.System.html
+[`QueryState`]: https://docs.rs/bevy/0.9.0/bevy/ecs/query/struct.QueryState.html
+[`SystemState`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/struct.SystemState.html
+[`Local`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/struct.Local.html
+
+## Bevy ECS Now Uses GATS!
+
+<div class="release-feature-authors">authors: @BoxyUwU</div>
+
+Rust 1.65.0 [stabilized GATs](https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html) (Generic Associated Types), which enabled us to significantly simplify Bevy ECS Query internals.
+
+For awhile now, Bevy ECS has been hacking around the lack of GATs with a complicated nest of traits (`WorldQuery`, `WorldQueryGats` (a "lack of real GATs hack" trait), and `Fetch`).
+
+In **Bevy 0.9**, we now have a single [`WorldQuery`] trait! This makes Bevy ECS much easier to maintain, extend, debug, document, and understand.
+
+[`WorldQuery`]: https://docs.rs/bevy/0.9.0/bevy/ecs/query/trait.WorldQuery.html
+
+## Derive Resource
+
+<div class="release-feature-authors">authors: @devil-ira</div>
```

Comment:
A large portion of the work was done by Alice. She definitely deserves credit :)